controller/src/beerocks/master/son_management.cpp

```diff
@@ -1952,6 +1994,11 @@ void son_management::handle_bml_message(Socket *sd,
             op_error_code == eChannelScanOperationCode::SUCCESS) {
             auto channel_pool_set =
                 std::unordered_set<uint8_t>(channel_pool, channel_pool + channel_pool_size);
+            if (is_scan_all_channels_request(channel_pool_set))
```

Comment:
1. you didn't add `{}` which is against our coding convention
2. in case of error you are still setting the DB! this part should be fixed similar to what you did in the `beerocks_message::ACTION_BML_CHANNEL_SCAN_START_SCAN_REQUEST` handling